exchanges/binanceus/binanceus.go

```diff
@@ -0,0 +1,1805 @@
+}
+
+// GetAllOCOOrder to retrieve all OCO orders based on provided optional parameters. Please note the maximum limit is 1,000 orders.
+func (b *Binanceus) GetAllOCOOrder(ctx context.Context, arg OCOOrdersRequestParams) ([]*OCOOrderResponse, error) {
+	params := url.Values{}
+	params.Set("timestamp", fmt.Sprint(time.Now().UnixMilli()))
+	var response []*OCOOrderResponse
+	if arg.FromID > 0 {
+		params.Set("fromId", fmt.Sprint(arg.FromID))
+	} else {
+		if arg.StartTime.Unix() > 0 && arg.StartTime.Before(arg.EndTime) {
+			params.Set("startTime", fmt.Sprint(arg.StartTime.UnixMilli()))
+			params.Set("endTime", fmt.Sprint(arg.EndTime.UnixMilli()))
+		} else if arg.StartTime.Unix() > 0 {
+			params.Set("startTime", fmt.Sprint(arg.StartTime.UnixMilli()))
+		}
+	}
+	if arg.Limit > 0 {
+		params.Set("limit", fmt.Sprint(arg.Limit))
+	}
+	if arg.RecvWindow > 0 {
+		params.Set("recvWindow", fmt.Sprint(arg.RecvWindow))
+	}
+	return response, b.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary,
+		http.MethodGet, ocoAllOrderList,
+		params, spotDefaultRate,
+		&response)
+}
+
+// GetOpenOCOOrders to query open OCO orders.
+func (b *Binanceus) GetOpenOCOOrders(ctx context.Context, RecvWindow uint) ([]*OCOOrderResponse, error) {
+	params := url.Values{}
+	params.Set("timestamp", fmt.Sprint(time.Now().UnixMilli()))
+	if RecvWindow > 0 {
+		params.Set("recvWindow", fmt.Sprint(RecvWindow))
+	} else {
+		params.Set("recvWindow", "30000")
+	}
+	var response []*OCOOrderResponse
+	return response, b.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary, http.MethodGet,
+		ocoOpenOrders, params,
+		spotDefaultRate, &response)
+}
+
+// CancelOCOOrder to cancel an entire order list.
+func (b *Binanceus) CancelOCOOrder(ctx context.Context, arg OCOOrdersDeleteRequestParams) (*OCOFullOrderResponse, error) {
+	var response OCOFullOrderResponse
+	params := url.Values{}
+	params.Set("timestamp", fmt.Sprint(time.Now().UnixMilli()))
+	if arg.Symbol == "" || len(arg.Symbol) <= 2 {
+		return nil, errIncompleteArguments
+	}
+	if arg.OrderListID > 0 {
+		params.Set("orderListId", fmt.Sprint(arg.OrderListID))
+	} else if arg.ListClientOrderID != "" {
+		params.Set("listClientOrderId", fmt.Sprint(arg.ListClientOrderID))
+	} else {
+		return nil, errIncompleteArguments
+	}
+	if arg.RecvWindow > 0 {
+		params.Set("recvWindow", fmt.Sprint(arg.RecvWindow))
+	}
+	return &response, b.SendAuthHTTPRequest(ctx, exchange.RestSpotSupplementary,
+		http.MethodGet, ocoOrderList,
+		params, spotDefaultRate, &response)
+}
+
+// OTC end points
+
+// GetSupportedCoinPairs  to get a list of supported coin pairs for convert.
+// returns list of CoinPairInfo
+func (b *Binanceus) GetSupportedCoinPairs(ctx context.Context, symbol currency.Pair) ([]*CoinPairInfo, error) {
+	params := url.Values{}
+	if symbol.Base.String() != "" && symbol.Quote.String() != "" {
```

Comment:
currency.Code has a method `IsEmpty()` please convert all comparisons to methods across this implementation. Instead of doing a string comparison.